### PR TITLE
update keras and add runtime Keras dependencies

### PR DESCRIPTION
**Pull request by rikenm1**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Add runtime dependencies of keras. 
Update keras to 3.1.1

###### Change Log  

- Change: Add python-namex, python-rich, python-poetry-core, python-lark, python-fastjsonschema, python-optree, python-markdown-it-py, python-mdurl
- Change: promote poetry to core
- Change: upgrade keras to 3.1.1

###### Does this affect the toolchain?  

**YES/NO**
NO

###### Test Methodology

- Pipeline build id: [539638](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=539638&view=results)

### CODE REVIEW

File: SPECS/python-optree/python-optree.spec [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+%files -n python3-optree
+%defattr(-,root,root)
+%license LICENSE
+%{python3_sitelib}/*
+
+%changelog
+* Fri Mar 29 2024 Riken Maharjan <rmaharjan@microsoft.com> - 0.11.0-1
+- Original version for CBL-Mariner.
```
Comment (nicogbg) [inline SPECS/python-optree/python-optree.spec:46]: Azure Linux instead of CBL-Mariner ?

File: SPECS/python-poetry-core/python-poetry-core.spec [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+%pyproject_save_files poetry
+
+
+%files -n python3-poetry-core -f %{pyproject_files}
+%doc README.md
+%license LICENSE
+
+%changelog
+* Fri Mar 29 2024 Riken Maharjan <rmaharjan@microsoft.com> - 1.9.0-1
+- Initial CBL-Mariner import from Fedora 40 (license: MIT).
```
Comment (nicogbg) [inline SPECS/python-poetry-core/python-poetry-core.spec:60]: Azure Linux ?

File: SPECS/python-rich/python-rich.spec [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+%pyproject_check_import
+%pytest -vv
+
+%files -n python3-rich -f %{pyproject_files}
+%license LICENSE
+%doc README.md
+
+%changelog
+* Mon Mar 28 2024 Riken Maharjan <rmaharjan@microsoft.com> - 13.7.1-2
+- Initial CBL-Mariner import from Fedora 41 (license: MIT).
```
Comment (nicogbg) [inline SPECS/python-rich/python-rich.spec:63]: Azure Linux ?

**Review by nicogbg — APPROVED** [review]

File: SPECS/python-poetry-core/python-poetry-core.spec (hunk 2 above)
Comment (rikenm1) [reply SPECS/python-poetry-core/python-poetry-core.spec]: updated

File: SPECS/python-rich/python-rich.spec (hunk 3 above)
Comment (rikenm1) [reply SPECS/python-rich/python-rich.spec]: updated

**Pull request merged**